Social Share Fixes #13

ProgrammerSONGBIT opened the pull request:
- Added **Twitter, LinkedIn, Copy Link** sharing buttons to Individual Stats page
- Beautiful **purple gradient UI** with hover animations & shine effects
- **Fully mobile responsive** (buttons stack on mobile) 
- social sharing

ProgrammerSONGBIT committed:
Implement share stats feature in StudentDashboard

Added functionality to share stats on Twitter, LinkedIn, and via copy link.

ProgrammerSONGBIT committed:
Refactor shareStats function and improve layout

git commit -m "feat: add social sharing buttons to student dashboard

- Add Twitter, LinkedIn, and Copy Link buttons to share stats

ProgrammerSONGBIT commented:
social share = studentdashboard typescript modified file

dipamsen commented:
- Copy link is not useful, as the link is specific to the logged in  user
- The styles don't match with the rest of the page.

ProgrammerSONGBIT committed:
Update StudentDashboard.tsx

**Changes:**
1.  **Removed copy link** - Now shares generic KWoC URL (not user-specific dashboard)
2.  **Removed custom styling** - Uses inline styles with Twitter/LinkedIn brand colors
3.  **Improved share text** - More engaging: "I just contributed to open source through KWoC 2025! X PRs merged, Y commits made. Join me! "
4.  **Better security** - Added `noopener,noreferrer` to window.open
5.  **Accessibility** - Added proper aria-labels
6.  **No CSS file needed** - Everything inline, matches page design

**Also fixed:**
- Added missing `key` props
- Changed `==` to `===`
- Added return types and number formatting

The share buttons now integrate seamlessly and are actually useful for students to share their achievements!

ProgrammerSONGBIT commented:
> * Copy link is not useful, as the link is specific to the logged in  user
> * The styles don't match with the rest of the page.

Fixed all issues and improved 6 feature
**Removed copy link**
**Removed custom styling** 
**Improved share text**
**Better security** - Added noopener,noreferrer to [window.open](http://window.open/) 
**Accessibility** 
**No CSS file needed**
